models/camp.js

```diff
@@ -37,7 +37,12 @@ var Camp = bookshelf.Model.extend({
 
         // let _camps_members = constants.CAMP_MEMBERS_TABLE_NAME;
         // let _users = constants.USERS_TABLE_NAME;
-        let query = "SELECT users.*,camp_members.status AS member_status,SUM(IF(tickets.ticket_id>0,1,0)) AS ticket_count,SUM(tickets.inside_event) AS inside_event, camp_members.addinfo_json AS camps_members_addinfo_json FROM users INNER JOIN camp_members on users.user_id=camp_members.user_id left join tickets on tickets.holder_id=users.user_id and tickets.event_id='" + this.attributes.event_id + "' where camp_members.camp_id=" + this.attributes.id + " group by users.user_id";
+        let query = `SELECT users.*, camp_members.status AS member_status, SUM(IF(tickets.ticket_id>0,1,0)) AS ticket_count, SUM(tickets.inside_event) AS inside_event, camp_members.addinfo_json AS camps_members_addinfo_json 
+        FROM users 
+            INNER JOIN camp_members on users.user_id=camp_members.user_id 
+            LEFT JOIN tickets on tickets.holder_id=users.user_id AND tickets.event_id='${this.attributes.event_id}' 
+            WHERE camp_members.camp_id=${this.attributes.id} GROUP BY users.user_id, member_status, camps_members_addinfo_json`;
+
         return knex //(_users)
```

Comment:
should fix following dump when editing a camp :
added GROUP BY to 'member_status', 'camps_members_addinfo_json' which are mapped 1:1 to every user_id in the join query.

07/01/18 02:25:35.847 ERROR app.js : Possibly Unhandled Rejection at: Promise
        {"isFulfilled":false,"isRejected":true,"rejectionReason":{"message":"SELECT users.*,camp_members.status AS member_status,SUM(IF(tickets.ticket_id>0,1,0)) AS ticket_count,SUM(tickets.inside_event) AS inside_event, camp_members.addinfo_json AS camps_members_addinfo_json FROM users INNER JOIN camp_members on users.user_id=camp_members.user_id left join tickets on tickets.holder_id=users.user_id and tickets.event_id='MIDBURN2018' where camp_members.camp_id=3 group by users.user_id - ER_WRONG_FIELD_WITH_GROUP: **Expression #31 of SELECT list is not in GROUP BY** clause and contains nonaggregated column '**spark.camp_members.status**' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by","stack":"Error: ER_WRONG_FIELD_WITH_GROUP: Expression #31 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'spark.camp_members.status' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by\n    at Query.Sequence._packetToError (/home/yaniv/workspace/Spark/node_modules/mysql/lib/protocol/sequences/Sequence.js:52:14)\n    at Query.ErrorPacket (/home/yaniv/workspace/Spark/node_modules/mysql/lib/protocol/sequences/Query.js:77:18)\n    at Protocol._parsePacket (/home/yaniv/workspace/Spark/node_modules/mysql/lib/protocol/Protocol.js:280:23)\n    at Parser.write (/home/yaniv/workspace/Spark/node_modules/mysql/lib/protocol/Parser.js:75:12)\n    at Protocol.write (/home/yaniv/workspace/Spark/node_modules/mysql/lib/protocol/Protocol.js:39:16)\n    at Socket.<anonymous> (/home/yaniv/workspace/Spark/node_modules/mysql/lib/Connection.js:103:28)\n    at emitOne (events.js:96:13)\n    at Socket.emit (events.js:191:7)\n    at readableAddChunk (_stream_readable.js:178:18)\n   at Socket.Readable.push (_stream_readable.js:136:10)\n    at TCP.onread (net.js:560:20)\n    --------------------\n    at Protocol._enqueue (/home/yaniv/workspace/Spark/node_modules/mysql/lib/protocol/Protocol.js:141:48)\n    at Connection.query (/home/yaniv/workspace/Spark/node_modules/mysql/lib/Connection.js:208:25)\n    at /home/yaniv/workspace/Spark/node_modules/knex/lib/dialects/mysql/index.js:152:18\n    at Promise._execute (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/debuggability.js:300:9)\n    at Promise._resolveFromExecutor (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/promise.js:483:18)\n    at new Promise (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/promise.js:79:10)\n    at Client_MySQL._query (/home/yaniv/workspace/Spark/node_modules/knex/lib/dialects/mysql/index.js:146:12)\n    at Client_MySQL.query (/home/yaniv/workspace/Spark/node_modules/knex/lib/client.js:197:17)\n    at Runner.<anonymous> (/home/yaniv/workspace/Spark/node_modules/knex/lib/runner.js:146:36)\n    at Runner.tryCatcher (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/util.js:16:23)\n    at Runner.query (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/method.js:15:34)\n    at /home/yaniv/workspace/Spark/node_modules/knex/lib/runner.js:65:21\n    at tryCatcher (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/util.js:16:23)\n    at /home/yaniv/workspace/Spark/node_modules/bluebird/js/release/using.js:185:26\n    at tryCatcher (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/util.js:16:23)\n    at Promise._settlePromiseFromHandler (/home/yaniv/workspace/Spark/node_modules/bluebird/js/release/promise.js:512:31)","code":"ER_WRONG_FIELD_WITH_GROUP","errno":1055,"sqlState":"42000","index":0}}